zerver/views/webhooks/wordpress.py

```diff
@@ -0,0 +1,52 @@
+# Webhooks for external integrations.
+from __future__ import absolute_import
+from django.utils.translation import ugettext as _
+from django.http import HttpRequest, HttpResponse
+from zerver.models import Client, get_client, UserProfile
+from zerver.lib.actions import check_send_message
+from zerver.lib.response import json_success, json_error
+from zerver.decorator import REQ, has_request_variables, api_key_only_webhook_view
+
+from six import text_type
+
+@api_key_only_webhook_view("Wordpress")
+@has_request_variables
+def api_wordpress_webhook(request, user_profile, client,
+                          stream=REQ(default="wordpress"),
+                          topic=REQ(default="WordPress Notification"),
+                          hook=REQ(default="WordPress Action"),
+                          post_title=REQ(default="New WordPress Post"),
+                          post_type=REQ(default="post"),
+                          post_url=REQ(default="WordPress Post URL"),
+                          display_name=REQ(default="New User Name"),
+                          user_email=REQ(default="New User Email"),
+                          user_login=REQ(default="Logged in User")):
+    # type: (HttpRequest, UserProfile, Client, text_type, text_type, text_type, text_type, text_type, text_type, text_type, text_type, text_type) -> HttpResponse
+
+    # remove trailing whitespace (issue for some test fixtures)
+    hook = hook.rstrip()
+    post_title = post_title.rstrip()
```

Comment:
I'm not sure which one you mean. hook is what identifies which WordPress action the user is using. WordPress will always send a value for it, even if the user supplies nothing else.

If it's last parameter in a test fixture (like publish_post_no_data_provided, where it is the only one) it will have the trailing newline. I tried removing the newline in the test fixture, but the linter didn't like that. So hook needs to be stripped or the code won't correctly recognize the action type when invoked from tests. (Or do it in every individual string comparison, which I find harder to read.)

The others may have newlines or not, depending on the order they appear in the text fixtures. Since I was stripping some (there was another case in some now removed code) I stripped all of them and then didn't include the trailing '\n' in the tests.

It's not a big deal to remove those and only strip hook. I'll just add the appropriate newlines back in the tests so their string comparisons work.